File: m3-h2/src/main/java/ru/yegorpilipenko/otus/spring/m3h2/controller/GenreController.java
package ru.yegorpilipenko.otus.spring.m3h2.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import ru.yegorpilipenko.otus.spring.m3h2.model.Genre;
import ru.yegorpilipenko.otus.spring.m3h2.service.GenreService;

import java.util.Collection;

import static ru.yegorpilipenko.otus.spring.m3h2.controller.GenreController.URL;

@RestController
@RequestMapping(URL)
@RequiredArgsConstructor
public final class GenreController {

    static final String URL = "/genres";

    private final GenreService genreService;

    @GetMapping
    public Collection<Genre> getAll() {
        return genreService.getAll();
    }

}

File: m2-h3/src/main/kotlin/ru/yegorpilipenko/otus/spring/m2h3/controller/GenreController.kt
package ru.yegorpilipenko.otus.spring.m2h3.controller

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import ru.yegorpilipenko.otus.spring.m2h3.service.GenreService

@Controller
class GenreController(val genreService: GenreService) {

    @GetMapping("/genres")
    fun genres(model: Model): String {
        val genres = genreService.getAll()
        model.addAttribute("genres", genres)
        return "genres"
    }

}
